source/jormungandr/jormungandr/interfaces/v1/serializer/JSONSchema.py

```diff
@@ -0,0 +1,165 @@
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+from serpy import fields, Serializer
+
+TYPE_MAP = {
+    'unknown': {
+        'type': 'unknown',
+    },
+    'method': {
+        'type': 'method',
+    },
+    'str': {
+        'type': 'string',
+    },
+    'float': {
+        'type': 'number'
+    },
+    'int': {
+        'type': 'integer'
+    },
+    'bool': {
+        'type': 'boolean',
+    },
+}
+
+class JSONSchema(Serializer):
+    properties = fields.MethodField('get_properties')
+    type = fields.MethodField('get_constant_object')
+    required = fields.MethodField('get_required')
+    # Weird name to ensure it will be processed a the end
+    _definitions = fields.MethodField('get_definitions', display_none=False, label='definitions')
+
+    def __init__(self, instance=None, many=False, data=None, context=None, root=False, **kwargs):
+        super(JSONSchema, self).__init__(instance, many, data, context, **kwargs)
+        self.root = root
+        self.definitions = {}
+
+    def get_constant_object(self, obj):
+        return 'object'
+
+    def get_definitions(self, obj):
+        if self.root:
+            properties = {}
+            definitions = self.definitions
+
+            while definitions:
+                field_name, field = definitions.popitem()
+                schema, nestedDefinitions = self.__class__._from_nested_schema(field)
+                properties[field_name] = schema
+                self.add_new_definitions(definitions, nestedDefinitions, properties.keys())
+
+            return properties
+        else:
+            return None
+
+    def add_new_definitions(self, definitionsToProcessed, newDefinitions, processedDefinitions):
+        reallyNewDefinitions = set(newDefinitions.keys()) - set(processedDefinitions)
+        for k in reallyNewDefinitions:
+            definitionsToProcessed.update({k: newDefinitions.get(k)})
+
+    def get_properties(self, obj):
+        mapping = {}
+        mapping[fields.StrField] = 'str'
+        mapping[fields.IntField] = 'int'
+        mapping[fields.FloatField] = 'float'
+        mapping[fields.BoolField] = 'bool'
+        # Map the 2 next fields to avoid exceptions for now
+        mapping[fields.Field] = 'str'
+        # mapping[fields.MethodField] = 'method'
```

Comment:
We call ```method.__name__ + '_jsonschema_pre_type_mapping' ``` to get a description of the output
like https://github.com/CanalTP/navitia/pull/2055/files#diff-8616a8e682cc2f9747d29e22bea695c7R73